sdk/test/AzureTest/Scrubbing.swift

```diff
@@ -0,0 +1,164 @@
+        }
+        return editedString
+    }
+    
+    static func scrubSubscriptionIDs(response: Foundation.URLResponse, data: Data?) -> URLRequest {
+        fatalError("Unimplemented")
+    }
+    
+    public static func scrubRequests(request: URLRequest) -> URLRequest {
+        var cleanedRequest = request
+        cleanedRequest = scrubSubscriptionIDs(request: cleanedRequest)
+        
+        return cleanedRequest
+    }
+    
+    static func scrubResponse(response: Foundation.URLResponse,data: Data?) -> (Foundation.URLResponse, Data?) {
+        fatalError("Unimplemented")
+    }
+}
+
+
+extension Array where Element == Any {
+    func dictionaryForIndex(_ index: Int) throws -> Dictionary<String,Any> {
+        if let dictionary = self[index] as? Dictionary<String, Any> {
+            return dictionary
+        }
+        throw "Not a dictionary"
+    }
+    
+    func arrayForIndex(_ index: Int) throws -> Array<Any> {
+        if let array = self[index] as? Array<Any> {
+            return array
+        }
+        throw "Not an array"
+    }
+    
+    func stringForIndex(_ index: Int) throws -> String {
+        if let string = self[index] as? String {
+            return string
+        }
+        throw "Not a string"
+    }
+    
+}
+
+ extension Dictionary where Key == String, Value == Any {
+    
+    static func findValue(key: String, dictionary: Dictionary<String,Any>) -> Any? {
+       if dictionary[key] != nil {
+           return dictionary[key]
+       }
+        
+       for testKey in dictionary.keys {
+           if let innerDictionary = dictionary[testKey] as? Dictionary<String , Any> {
+               let returnedValue = findValue(key: key, dictionary: innerDictionary)
+               if returnedValue != nil {
+                   return returnedValue
+               }
+           }
+       }
+       return nil
+    }
+    
+    func dicionaryForKey(_ key: Key) throws -> Dictionary<String,Any>  {
+        if let innerDictionary = self[key] as? Dictionary<String,Any> {
+            return innerDictionary
+        }
+        throw "Not a dictionary"
+    }
+    
+    func arrayForKey(_ key: Key) throws -> Array<Any>  {
+        if let innerArray = self[key] as? Array<Any> {
+            return innerArray
+        }
+        throw "Not an array"
+    }
+    
+    func stringForKey(_ key: Key) throws -> String {
+        if let string = self[key] as? String {
+            return string
+        }
+        throw "Not a string"
+    }
+}
+
+extension String {
+    public func dictionaryFromString() throws -> Dictionary<String, Any> {
+        do {
+            let data = self.data(using: .utf8)
+            let dictionary = try JSONSerialization.jsonObject(with: data!, options: []) as? [String: Any]
+            return dictionary!
+        } catch {
+            throw "Not a dictionary"
+        }
```

Comment:
It's a little weird to throw but also use forced-unwrapping here. If dictionary is `nil` you should also throw "Not a dictionary" instead of crashing due to the force-unwrap.